common/functions.rb

```diff
@@ -84,14 +84,31 @@ def dim_full_indices(x, dim_no, idxs)
   ind
 end
 
+# Calls `paired_access_idxs` and returns the corresponding elements from `x`.
+#
+# @see paired_access_idxs
+#
+# @param x     [Numo::NArray] Matrix.
+# @param idxs1 [Array-like] indexes in the 1st dimension.
+# @param idxs2 [Array-like] indexes in the 2nd dimension.
+# @return [Numo::NArray] Elements at the given indexes.
 def paired_access(x, idxs1, idxs2)
   x[paired_access_idxs(x, idxs1, idxs2)]
 end
 
+# Return the full indexes when given two arrays where each element with the
+# same index corresponding to the 1st and 2nd index in the matrix `x`.
+# This is the same as numpy's notation of `x[[0, 1, 2], [2, 2, 5]]` meaning
+# the elements at [0, 2], [1, 2], [2, 5].
+#
+# @param x     [Numo::NArray] Matrix.
+# @param idxs1 [Array-like] indexes in the 1st dimension.
+# @param idxs2 [Array-like] indexes in the 2nd dimension.
+# @return [Array<Integer>] Array of the indices.
 def paired_access_idxs(x, idxs1, idxs2)
   tmp_idxs2 = idxs2.to_a
   full_idxs = idxs1.to_a.map.with_index do |val, i|
-    x.shape[0] * val + tmp_idxs2[i]
+    x.shape[1] * val + tmp_idxs2[i]
```

Comment:
The index calculation was wrong as I was adding the width of the 1st dimension, not the 2nd!